src/live.js

```diff
@@ -1,69 +1,61 @@
 'use strict';
 var rc_util = require('rippled-network-crawler/src/lib/utility.js');
 var _ = require('lodash');
-var writeToGraphite = require('./lib/report_graphite.js').writeToGraphite;
+var writeToGraphite = require('./lib/report_graphite');
+var sqs_u = require('./lib/sqs_util');
 var moment = require('moment');
 var Promise = require('bluebird');
 var graphite = require('graphite');
-var graphiteClient;
-var lastCrawlID;
 
-function advanceLastCrawlID(batchSize, ids) {
-  ids.start = lastCrawlID + 1;
-  ids.end = lastCrawlID + batchSize;
-  lastCrawlID = ids.end;
-}
+module.exports = function(max, timeout, queueUrl, dbUrl, graphiteUrl) {
+  var graphiteClient = graphite.createClient(graphiteUrl);
+  var logsql = false;
+  var count = 0;
 
-function reportCrawl(crawl) {
-  writeToGraphite(crawl, graphiteClient);
-  return parseInt(crawl.id, 10);
-}
+  function processCrawl(crawl) {
+    writeToGraphite(crawl, graphiteClient)
+  }
 
-function recReport(ids, batchSize, dbUrl, logsql) {
-  rc_util
-  .getRowsByIds(dbUrl, ids.start, ids.end, logsql)
-  .then(function(latestCrawls) {
-    if (latestCrawls && latestCrawls.length) {
-      console.log('received crawls [%d, %d]',
-        ids.start, ids.start + latestCrawls.length - 1);
-      ids.start += latestCrawls.length || 0;
-      _.map(latestCrawls, reportCrawl);
-    }
-    if (ids.start > ids.end) {
-      advanceLastCrawlID(batchSize, ids);
-    }
-    process.nextTick(function() {
-      recReport(ids, batchSize, dbUrl, logsql);
+  function processMessage(workerID) {
+    sqs_u.getMessage(queueUrl)
+    .then(function(response) {
+      if (response.Messages) {
+        var id = response.Messages[0].Body;
+        var receiptHandle = response.Messages[0].ReceiptHandle;
+        rc_util
+        .getRowById(dbUrl, id, logsql)
+        .then(function(crawl) {
+          if (!crawl) {
+            console.error('No crawls with id %s', id);
+            count -= 1;
+            return -1;
+          }
+
+          processCrawl(crawl);
+          count -= 1;
+
+        }).then(function() {
```

Comment:
Is there a scenario where `processCrawl` fails and we wouldn't want to delete the message?